rllib/policy/tests/test_sample_batch.py

```diff
@@ -341,6 +346,76 @@ def test_shuffle_with_interceptor(self):
         # but have also been shuffled.
         check(s["a"], [2, 3, 4, 3, 4, 5, 4, 5, 6, 5, 6, 7, 6, 7, 8], false=True)
 
+    def test_to_device(self):
+        """Tests whether to_device works properly under different circumestances"""
+        torch, _ = try_import_torch()
+
+        # sample batch includes
+        #   a numpy array (a)
+        #   a nested stucture of dict, tuple and lists (b) of numpys and None
+        #   info dict
+        #   a nested structure that ends up with tensors and ints(c)
+        #   a tensor with float64 values (d)
+        #   a float64 tensor with possibly wrong device (depends on if cuda available)
+        #   repeated value object with np.array leaves (f)
+
+        cuda_available = int(os.environ.get("RLLIB_NUM_GPUS", "0")) > 0
+        cuda_if_possible = torch.device("cuda:0" if cuda_available else "cpu")
+        s = SampleBatch(
+            {
+                "a": np.array([1, 2]),
+                "b": {"c": (np.array([4, 5]), np.array([5, 6])), "e": [{"f": None}]},
+                "c": {"d": torch.Tensor([1, 2]), "g": (torch.Tensor([3, 4]), 1)},
+                "d": torch.Tensor([1.0, 2.0]).double(),
+                "e": torch.Tensor([1.0, 2.0]).double().to(cuda_if_possible),
+                "f": RepeatedValues(np.array([[1, 2, 0, 0]]), lengths=[2], max_len=4),
+                SampleBatch.SEQ_LENS: np.array([2, 3, 1]),
+                "state_in_0": np.array([1.0, 3.0, 4.0]),
+                SampleBatch.INFOS: np.array([{"a": 1}, {"b": 2}, {"c": 3}]),
```

Comment:
INFOS's dtype is object right? and we basically don't do anything to it if I understand correctly.

Reply:
Yes. correct.